File: TrainWebUI/src/types/payment.ts
import { BookingDto } from './booking';

export enum PaymentMethod {
  Visa = 0,
  Momo = 1,
  VnPay = 2
}

export enum PaymentStatus {
  Success = 0,
  Pending = 1,
  Failed = 2
}

export interface PaymentEntity {
  id: string;
  bookingId?: string; // legacy
  booking?: BookingDto; // backend returns nested booking dto
  amount: number;
  method: PaymentMethod;
  status: PaymentStatus;
  createdAt: string;
}

export interface PaymentRequest {
  booking: { id: string };
  amount?: number;
  method: PaymentMethod;
  status: PaymentStatus;
}

export const PAYMENT_METHOD_LABELS: Record<PaymentMethod, string> = {
  [PaymentMethod.Visa]: 'Visa',
  [PaymentMethod.Momo]: 'Momo',
  [PaymentMethod.VnPay]: 'VnPay'
};

export const PAYMENT_STATUS_LABELS: Record<PaymentStatus, string> = {
  [PaymentStatus.Success]: 'Thành công',
  [PaymentStatus.Pending]: 'Đang xử lý',
  [PaymentStatus.Failed]: 'Thất bại'
};

export function isPaymentMethod(value: number): value is PaymentMethod {
  return value >= 0 && value <= 2;
}

export function isPaymentStatus(value: number): value is PaymentStatus {
  return value >= 0 && value <= 2;
}

File: TrainWeb.Domain/Domain/Payment.cs
using TrainWeb.Domain.Enum;

namespace TrainWeb.Domain.Domain
{
    public class Payment
    {
        public string? Id { get; }

        public Booking? Booking { get; init; }   

        public string? BookingId { get; init; }  

        public double Amount { get; set; }

        public PaymentMethod Method { get; set; } = PaymentMethod.Visa;

        public PaymentStatus Status { get; set; } = PaymentStatus.Pending;

        public DateTime CreatedAt { get; init; } = DateTime.UtcNow;

        public Payment(
            string? id,
            Booking? booking,
            string? bookingId,
            double amount,
            PaymentMethod method,
            PaymentStatus status,
            DateTime createdAt
        )
        {
            Id = id;
            Booking = booking;

            BookingId = !string.IsNullOrWhiteSpace(bookingId) ? bookingId : booking?.Id;

            Amount = amount;
            Method = method;
            Status = status;
            CreatedAt = createdAt;
        }
    }
}
